applications/crossbar/src/modules/cb_cdrs.erl

```diff
@@ -594,7 +613,145 @@ load_cdr(CDRId, Context) ->
     lager:debug("error loading cdr by id ~p", [CDRId]),
     crossbar_util:response('error', <<"could not find cdr with supplied id">>, 404, Context).
 
--ifdef(TEST).
--include_lib("eunit/include/eunit.hrl").
+-spec maybe_group_cdrs(cb_context:context(), JObjs) ->
+                              wh_json:object() | JObjs.
+maybe_group_cdrs(Context, JObjs) ->
+    case wh_util:is_true(cb_context:req_value(Context, <<"grouped">>)) of
+        'true' ->
+            group_cdrs(JObjs, fun(JObj) -> normalize_cdr(JObj, Context) end);
+        'false' -> JObjs
+    end.
+
+-record(group_value, {timestamp :: gregorian_seconds()
+                      ,call_id :: ne_binary()
+                      ,cdr :: kzd_cdr:doc()
+                     }).
+
+-type group_value() :: #group_value{}.
+-type group_values() :: [group_value(),...] | [].
 
+-type transform_fun() :: fun((wh_json:object()) -> wh_json:object()).
+
+-ifdef(TEST).
+-spec group_cdrs(wh_json:objects()) ->
+                        wh_json:object().
+group_cdrs(JObjs) ->
+    group_cdrs(JObjs, fun wh_util:identity/1).
 -endif.
```

Comment:
Ah I see. Well maybe this should be inside the `test/` and calling `group_cdrs/2` from there?